Randomise the order in which player's actions are updated - potential first turn fix

ScrambledEggsOnToast opened the pull request:
The idea behind this is to have the game process each player's actions in a different order each turn. This way first turn is randomised, and same turn wonders are allocated randomly. NOTE: this is untested - it is literally an idea that I just smacked out.

ScrambledEggsOnToast commented:
By the way, I've tested this for a couple of turns with AIs, and it seems stable. It's hard to tell if it actually does as intended without testing with humans though.

nik9000 commented on CvGameCoreDLL_Expansion2/CvGame.cpp:
```diff
@@ -8354,9 +8368,23 @@ void CvGame::updateMoves()
+			{
+				processOrder[Oi] = Oi;
+			}
+			for (Oi = 0; Oi < MAX_PLAYERS - 1; Oi++) 
```
I'd yank this into a shuffle method. Personally I'd implement it just like https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle#The_modern_algorithm and comment about it being a Fisher-Yates shuffle. Shuffling is notoriously easy to screw up.

ScrambledEggsOnToast commented:
I think it already is Fisher-Yates. We have to generate the array `[0,1,2,...]` first - that's the first for loop.  The second for loop is the actual shuffle of that array. The last is the original for loop, just with the random order substituted in. Commenting sounds like a good idea. I'm not sure about putting it in it's own method. It's only really used in two places.

ScrambledEggsOnToast commented:
As Delnar_Ersike has pointed out, there actually already is a function for shuffling an array in this way: `shuffleArray` over in `cvgamecoreutils.cpp`. I've switched over to use of this function. Also, I've added a 750ms delay between activation of players' turns, to help with fixing effects of response time in first-turn.

ScrambledEggsOnToast commented:
After further discussion with Delnar_Ersike, it looks like this will require a bit more thought.
